File: ios/Sasquatsh/Views/Invitations/InvitationAcceptView.swift
import SwiftUI

struct InvitationAcceptView: View {
    let inviteCode: String
    @Environment(\.services) private var services
    @Environment(\.dismiss) private var dismiss
    @State private var vm = InvitationViewModel()

    var body: some View {
        NavigationStack {
            VStack(spacing: 24) {
                if vm.isLoading && vm.gameInvitation == nil && vm.successMessage == nil {
                    LoadingView(message: "Loading invitation...")
                } else if let msg = vm.successMessage {
                    VStack(spacing: 16) {
                        Image(systemName: "checkmark.circle.fill")
                            .font(.system(size: 60))
                            .foregroundStyle(.green)
                        Text("Invitation Accepted!")
                            .font(.md3HeadlineMedium)
                        Text(msg)
                            .font(.md3BodyMedium)
                            .foregroundStyle(Color.md3OnSurfaceVariant)
                            .multilineTextAlignment(.center)
                        Button {
                            dismiss()
                        } label: {
                            Text("Done")
                                .primaryButtonStyle()
                        }
                        .padding(.horizontal, 40)
                    }
                } else if let error = vm.error {
                    VStack(spacing: 16) {
                        Image(systemName: "exclamationmark.triangle.fill")
                            .font(.system(size: 60))
                            .foregroundStyle(.orange)
                        Text("Could not load invitation")
                            .font(.md3HeadlineSmall)
                        Text(error)
                            .font(.md3BodyMedium)
                            .foregroundStyle(Color.md3OnSurfaceVariant)
                            .multilineTextAlignment(.center)
                    }
                } else if let invitation = vm.gameInvitation, let event = invitation.event {
                    VStack(spacing: 20) {
                        Image(systemName: "envelope.open.fill")
                            .font(.system(size: 48))
                            .foregroundStyle(Color.md3Primary)

                        Text("You're Invited!")
                            .font(.md3HeadlineMedium)

                        VStack(spacing: 8) {
                            Text(event.title)
                                .font(.md3TitleMedium)
                                .foregroundStyle(Color.md3OnSurface)

                            if let host = event.host?.displayName {
                                Text("Hosted by \(host)")
                                    .font(.md3BodyMedium)
                                    .foregroundStyle(Color.md3OnSurfaceVariant)
                            }

                            HStack(spacing: 12) {
                                Label(event.eventDate.toDate?.displayDate ?? event.eventDate, systemImage: "calendar")
                                Label(event.startTime.to12HourTime, systemImage: "clock")
                            }
                            .font(.md3BodySmall)

                            if let city = event.city, let state = event.state {
                                Label("\(city), \(state)", systemImage: "mappin")
                                    .font(.md3BodySmall)
                            }
                        }
                        .padding()
                        .cardStyle()

                        Button {
                            Task { await vm.acceptGameInvitation(code: inviteCode) }
                        } label: {
                            Text("Accept Invitation")
                                .primaryButtonStyle()
                        }
                        .padding(.horizontal)
                    }
                }

                Spacer()
            }
            .padding()
            .navigationTitle("Game Invitation")
            .navigationBarTitleDisplayMode(.inline)
            .toolbar {
                ToolbarItem(placement: .cancellationAction) {
                    Button("Close") { dismiss() }
                }
            }
            .task {
                vm.configure(services: services)
                await vm.loadGameInvitation(code: inviteCode)
            }
        }
    }
}

File: android/app/src/main/java/com/sasquatsh/app/views/invitations/InvitationAcceptView.kt
package com.sasquatsh.app.views.invitations

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.CalendarMonth
import androidx.compose.material.icons.filled.CheckCircle
import androidx.compose.material.icons.filled.Close
import androidx.compose.material.icons.filled.LocationOn
import androidx.compose.material.icons.filled.Mail
import androidx.compose.material.icons.filled.Schedule
import androidx.compose.material.icons.filled.Warning
import androidx.compose.material3.Button
import androidx.compose.material3.Card
import androidx.compose.material3.CardDefaults
import androidx.compose.material3.ExperimentalMaterial3Api
import androidx.compose.material3.Icon
import androidx.compose.material3.IconButton
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Scaffold
import androidx.compose.material3.Text
import androidx.compose.material3.TopAppBar
import androidx.compose.runtime.Composable
import androidx.compose.runtime.LaunchedEffect
import androidx.compose.runtime.collectAsState
import androidx.compose.runtime.getValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.hilt.navigation.compose.hiltViewModel
import com.sasquatsh.app.viewmodels.InvitationViewModel
import com.sasquatsh.app.views.shared.LoadingView

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun InvitationAcceptView(
    inviteCode: String,
    onDismiss: () -> Unit,
    viewModel: InvitationViewModel = hiltViewModel()
) {
    val uiState by viewModel.uiState.collectAsState()

    LaunchedEffect(inviteCode) {
        viewModel.loadGameInvitation(inviteCode)
    }

    Scaffold(
        topBar = {
            TopAppBar(
                title = { Text("Game Invitation") },
                navigationIcon = {
                    IconButton(onClick = onDismiss) {
                        Icon(Icons.Default.Close, contentDescription = "Close")
                    }
                }
            )
        }
    ) { padding ->
        Column(
            modifier = Modifier
                .fillMaxSize()
                .padding(padding)
                .padding(16.dp),
            horizontalAlignment = Alignment.CenterHorizontally
        ) {
            when {
                uiState.isLoading && uiState.gameInvitation == null && uiState.successMessage == null -> {
                    LoadingView(message = "Loading invitation...")
                }

                uiState.successMessage != null -> {
                    Spacer(modifier = Modifier.height(40.dp))
                    Icon(
                        Icons.Default.CheckCircle,
                        contentDescription = null,
                        modifier = Modifier.size(60.dp),
                        tint = Color(0xFF4CAF50)
                    )
                    Spacer(modifier = Modifier.height(16.dp))
                    Text(
                        "Invitation Accepted!",
                        style = MaterialTheme.typography.headlineMedium
                    )
                    Spacer(modifier = Modifier.height(8.dp))
                    Text(
                        uiState.successMessage!!,
                        style = MaterialTheme.typography.bodyMedium,
                        color = MaterialTheme.colorScheme.onSurfaceVariant,
                        textAlign = TextAlign.Center
                    )
                    Spacer(modifier = Modifier.height(24.dp))
                    Button(
                        onClick = onDismiss,
                        modifier = Modifier
                            .fillMaxWidth()
                            .padding(horizontal = 40.dp)
                    ) {
                        Text("Done")
                    }
                }

                uiState.error != null -> {
                    Spacer(modifier = Modifier.height(40.dp))
                    Icon(
                        Icons.Default.Warning,
                        contentDescription = null,
                        modifier = Modifier.size(60.dp),
                        tint = Color(0xFFFFA500)
                    )
                    Spacer(modifier = Modifier.height(16.dp))
                    Text(
                        "Could not load invitation",
                        style = MaterialTheme.typography.headlineSmall
                    )
                    Spacer(modifier = Modifier.height(8.dp))
                    Text(
                        uiState.error!!,
                        style = MaterialTheme.typography.bodyMedium,
                        color = MaterialTheme.colorScheme.onSurfaceVariant,
                        textAlign = TextAlign.Center
                    )
                }

                uiState.gameInvitation != null -> {
                    val invitation = uiState.gameInvitation!!
                    val event = invitation.event

                    Spacer(modifier = Modifier.height(24.dp))

                    Icon(
                        Icons.Default.Mail,
                        contentDescription = null,
                        modifier = Modifier.size(48.dp),
                        tint = MaterialTheme.colorScheme.primary
                    )
                    Spacer(modifier = Modifier.height(16.dp))
                    Text(
                        "You're Invited!",
                        style = MaterialTheme.typography.headlineMedium
                    )

                    Spacer(modifier = Modifier.height(20.dp))

                    if (event != null) {
                        Card(
                            modifier = Modifier.fillMaxWidth(),
                            colors = CardDefaults.cardColors(
                                containerColor = MaterialTheme.colorScheme.surface
                            )
                        ) {
                            Column(modifier = Modifier.padding(16.dp)) {
                                Text(
                                    event.title,
                                    style = MaterialTheme.typography.titleMedium,
                                    color = MaterialTheme.colorScheme.onSurface
                                )

                                event.host?.displayName?.let { host ->
                                    Spacer(modifier = Modifier.height(4.dp))
                                    Text(
                                        "Hosted by $host",
                                        style = MaterialTheme.typography.bodyMedium,
                                        color = MaterialTheme.colorScheme.onSurfaceVariant
                                    )
                                }

                                Spacer(modifier = Modifier.height(8.dp))

                                Row(horizontalArrangement = Arrangement.spacedBy(12.dp)) {
                                    Row(verticalAlignment = Alignment.CenterVertically) {
                                        Icon(
                                            Icons.Default.CalendarMonth,
                                            contentDescription = null,
                                            modifier = Modifier.size(14.dp),
                                            tint = MaterialTheme.colorScheme.onSurfaceVariant
                                        )
                                        Spacer(modifier = Modifier.width(4.dp))
                                        Text(
                                            event.eventDate,
                                            style = MaterialTheme.typography.bodySmall
                                        )
                                    }
                                    Row(verticalAlignment = Alignment.CenterVertically) {
                                        Icon(
                                            Icons.Default.Schedule,
                                            contentDescription = null,
                                            modifier = Modifier.size(14.dp),
                                            tint = MaterialTheme.colorScheme.onSurfaceVariant
                                        )
                                        Spacer(modifier = Modifier.width(4.dp))
                                        Text(
                                            event.startTime,
                                            style = MaterialTheme.typography.bodySmall
                                        )
                                    }
                                }

                                if (event.city != null && event.state != null) {
                                    Spacer(modifier = Modifier.height(4.dp))
                                    Row(verticalAlignment = Alignment.CenterVertically) {
                                        Icon(
                                            Icons.Default.LocationOn,
                                            contentDescription = null,
                                            modifier = Modifier.size(14.dp),
                                            tint = MaterialTheme.colorScheme.onSurfaceVariant
                                        )
                                        Spacer(modifier = Modifier.width(4.dp))
                                        Text(
                                            "${event.city}, ${event.state}",
                                            style = MaterialTheme.typography.bodySmall
                                        )
                                    }
                                }
                            }
                        }
                    }

                    Spacer(modifier = Modifier.height(24.dp))

                    Button(
                        onClick = { viewModel.acceptGameInvitation(inviteCode) },
                        modifier = Modifier
                            .fillMaxWidth()
                            .padding(horizontal = 16.dp)
                    ) {
                        Text("Accept Invitation")
                    }
                }
            }

            Spacer(modifier = Modifier.weight(1f))
        }
    }
}
